python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -0,0 +1,119 @@
+
+
+class OpenAIChatCompletionBase(OpenAIHandler, ChatCompletionClientBase):
+    async def complete_chat_async(
+        self,
+        messages: List[Dict[str, str]],
+        settings: "ChatRequestSettings",
+        logger: Optional[Logger] = None,
+    ) -> Union[str, List[str]]:
+        """Executes a chat completion request and returns the result.
+
+        Arguments:
+            messages {List[Tuple[str,str]]} -- The messages to use for the chat completion.
+            settings {ChatRequestSettings} -- The settings to use for the chat completion request.
+            logger {Optional[Logger]} -- The logger instance to use. (Optional)
+
+        Returns:
+            Union[str, List[str]] -- The completion result(s).
+        """
+        response = await self._send_request(
+            messages=messages, request_settings=settings, stream=False
+        )
+
+        if len(response.choices) == 1:
+            return response.choices[0].message.content
+        else:
+            return [choice.message.content for choice in response.choices]
+
+    async def complete_chat_with_functions_async(
+        self,
+        messages: List[Dict[str, str]],
+        functions: List[Dict[str, Any]],
+        request_settings: "ChatRequestSettings",
+        logger: Optional[Logger] = None,
+    ) -> Union[
+        Tuple[Optional[str], Optional[FunctionCall]],
+        List[Tuple[Optional[str], Optional[FunctionCall]]],
+    ]:
+        # TODO: tracking on token counts/etc.
+
+        response = await self._send_request(
+            messages=messages,
+            request_settings=request_settings,
+            stream=False,
+            functions=functions,
+        )
+
+        if len(response.choices) == 1:
+            return _parse_message(response.choices[0].message, self.log)
+        else:
+            return [
+                _parse_message(choice.message, self.log) for choice in response.choices
+            ]
+
+    async def complete_chat_stream_async(
+        self,
+        messages: List[Dict[str, str]],
+        settings: "ChatRequestSettings",
+        logger: Optional[Logger] = None,
+    ) -> AsyncGenerator[Union[str, List[str]], None]:
+        """Executes a chat completion request and returns the result.
+
+        Arguments:
+            messages {List[Tuple[str,str]]} -- The messages to use for the chat completion.
+            settings {ChatRequestSettings} -- The settings to use for the chat completion request.
+            logger {Optional[Logger]} -- The logger instance to use. (Optional)
+
+        Returns:
+            Union[str, List[str]] -- The completion result(s).
+        """
+        response = await self._send_request(
+            messages=messages, request_settings=settings, stream=True
+        )
+
+        # parse the completion text(s) and yield them
+        async for chunk in response:
+            text, index = self._parse_choices(chunk)
```

Comment:
Based on the following PR, we'd need to make some updated to handle parsing choices for streaming respsonses: https://github.com/microsoft/semantic-kernel/pull/3651.